src/modules/apps/e2e/report.tsx

```diff
@@ -0,0 +1,270 @@
+import { parseAppId } from '@vtex/api'
+
+import { SpecReport, AppReport, TestReport, SpecTestReport, Screenshot } from '../../../clients/Tester'
+import { useInterval } from './useInterval'
+
+type AppIdProps = {
+  appId: string
+}
+
+const AppId: React.FunctionComponent<AppIdProps> = ({ appId }) => {
+  const { name, version, build } = parseAppId(appId)
+  const [vendor, app] = name.split('.')
+  return (
+    <Box>
+      <Color blue>{vendor}</Color>
+      <Color dim>.</Color>
+      {`${app}@${version}`}
+      {build && <Color dim>{`-${build}`}</Color>}
+    </Box>
+  )
+}
+
+type SpecDetailProps = {
+  label: string
+  text: string
+  indented: boolean
+}
+
+const FailedSpecDetail: React.FunctionComponent<SpecDetailProps> = ({ label, text, indented }) => {
+  const nLines = text.split('\n')
+  return (
+    <Box flexDirection="column" marginBottom={1}>
+      <Text>
+        <Text underline>{`${label}:`}</Text>
+        {!indented && ` ${text}`}
+      </Text>
+      {indented && (
+        <Box marginTop={1}>
+          <Box flexDirection="column">{nLines.map(() => '│ ')}</Box>
+          <Box>{text}</Box>
+        </Box>
+      )}
+    </Box>
+  )
+}
+
+type SpecProps = {
+  spec: string
+  report: SpecReport
+}
+
+const FailedSpec: React.FunctionComponent<SpecProps> = ({ spec, report }) => {
+  const video = path<string>(['report', 'video'], report)
+  const screenshots = path<Screenshot[]>(['report', 'screenshots'], report)
+  return (
+    <Box flexDirection="column">
+      <Color bold>{`${spec}:`}</Color>
+      <Box flexDirection="column" marginLeft={2}>
+        {report.error && <FailedSpecDetail label={'Error'} text={report.error} indented={true} />}
+        {...pathOr<SpecTestReport[], SpecTestReport[]>([] as SpecTestReport[], ['report', 'tests'], report)
+          .filter(({ state }) => state !== 'passed')
+          .map(({ testId, title, body, stack, error }) => {
+            const _screenshots = screenshots.filter(ss => ss.testId === testId)
+            return (
+              <Box key={title.join('')} flexDirection="column">
+                <FailedSpecDetail label={'Test'} text={title.join(' ')} indented={false} />
+                {body && <FailedSpecDetail label={'Body'} text={body} indented={true} />}
+                {stack && <FailedSpecDetail label={'Stack'} text={stack} indented={true} />}
+                {error && <FailedSpecDetail label={'Error'} text={error} indented={true} />}
+                {_screenshots.length > 0 && (
+                  <FailedSpecDetail
+                    label={'Screenshots'}
+                    text={_screenshots.map(ss => ' ' + ss.path).join('\n')}
+                    indented={_screenshots.length > 1}
+                  />
+                )}
+              </Box>
+            )
+          })}
+        {video && <FailedSpecDetail label={'Video'} text={video} indented={false} />}
+      </Box>
+    </Box>
+  )
+}
```

Comment:
How about something like:
```ts
const FailedSpec: React.FunctionComponent<SpecProps> = ({ spec, report }) => {
  const video = report.report?.video
  const screenshots = report.report?.screenshots
  const notPassedSpecs = (report.report?.tests ?? []).filter(({ state }) => state !== 'passed')

  const errorsVisualization = notPassedSpecs.map(({ title, testId, body, error, stack }) => {
    const testScreenshots = screenshots.filter(curScreenshot => curScreenshot.testId === testId)
    return (
      <ErrorVisualization title={title} body={body} error={error} stack={stack} testScreenshots={testScreenshots} />
    )
  })

  return (
    <Box flexDirection="column">
      <Color bold>{`${spec}:`}</Color>
      <Box flexDirection="column" marginLeft={2}>
        {report.error && <FailedSpecDetail label={'Error'} text={report.error} indented={true} />}
        {errorsVisualization}
        {video && <FailedSpecDetail label={'Video'} text={video} indented={false} />}
      </Box>
    </Box>
  )
}

interface ErrorVisualizationProps {
  title: SpecTestReport['title']
  body: SpecTestReport['body']
  error: SpecTestReport['error']
  stack: SpecTestReport['stack']

  testScreenshots: Screenshot[]
}

const ErrorVisualization: React.FunctionComponent<ErrorVisualizationProps> = ({
  title,
  body,
  error,
  stack,
  testScreenshots,
}) => {
  const testScreenshotsText = testScreenshots.map(curScreenshot => ' ' + curScreenshot.path).join('\n')
  return (
    <Box key={title.join('')} flexDirection="column">
      <FailedSpecDetail label={'Test'} text={title.join(' ')} indented={false} />
      {body && <FailedSpecDetail label={'Body'} text={body} indented={true} />}
      {stack && <FailedSpecDetail label={'Stack'} text={stack} indented={true} />}
      {error && <FailedSpecDetail label={'Error'} text={error} indented={true} />}
      {testScreenshots.length > 0 && (
        <FailedSpecDetail label={'Screenshots'} text={testScreenshotsText} indented={testScreenshots.length > 1} />
      )}
    </Box>
  )
}

```

Reply:
Oh, I think I know what is the problem, when you create an array of elements in react you have to give each one a differente `key`:
```ts
  const errorsVisualization = notPassedSpecs.map(({ title, testId, body, error, stack }, index) => {
    const testScreenshots = screenshots.filter(curScreenshot => curScreenshot.testId === testId)
    return (
      <ErrorVisualization key={index} title={title} body={body} error={error} stack={stack} testScreenshots={testScreenshots} />
    )
  })

```

Reply:
Maybe this solves the problem?

Reply:
It worked!